client/signup/steps/plans/index.jsx

```diff
@@ -33,25 +33,25 @@ export class PlansStep extends Component {
 		experimentLoaded: false,
 	};
 
-	componentWillMount() {
+	componentDidMount() {
+		this.unsubscribe = subscribeIsDesktop( ( matchesDesktop ) =>
+			this.setState( { isDesktop: matchesDesktop } )
+		);
+		this.props.saveSignupStep( { stepName: this.props.stepName } );
+
 		if ( this.props.flowName === 'onboarding' || this.props.flowName === 'launch-site' ) {
-			loadExperimentAssignment( 'calypso_signup_monthly_plans_default_202201_v1' ).then(
+			loadExperimentAssignment( 'calypso_signup_monthly_plans_default_202201_v2' ).then(
```

Comment:
Should loading the experiment assignment also be cleaned up after on unmount? Won't it leak if the component gets unmounted during loading of the experiment assignment?

Reply:
I've refactored to remove the `setState()` on mount and use the `ProvideExperimentData` component. Hopefully this addresses your concern.